substratevm/src/org.graalvm.polyglot.nativeapi/src/org/graalvm/polyglot/nativeapi/PolyglotNativeAPI.java

```diff
@@ -138,10 +142,15 @@
         }
     }
 
+    private static class ErrorStateHolder {
+        public ErrorInfoHolder errorInfoHolder = null;
```

Comment:
Can we inline this into ErrorState?

Reply:
We could. The reason it was done this way was to make it possible for the ErrorInfoHolder to be NULL. It was originally inline, but it required putting a C null pointer for the ErrorInfoHolder.info, and a placeholder CCharPointerHolder. Is it ok for CCharPointerHolder to hold a null pointer or a dummy empty string?

Reply:
It should be fine, don't see a problem with that.

Reply:
ErrorStateHolder was made inline.